crates/biome_js_analyze/src/lint/nursery/use_error_cause.rs

```diff
@@ -0,0 +1,214 @@
+use biome_analyze::{
+    Ast, Rule, RuleDiagnostic, RuleSource, context::RuleContext, declare_lint_rule,
+};
+use biome_console::markup;
+use biome_js_syntax::{AnyJsBindingPattern, JsCatchClause, JsThrowStatement};
+use biome_rowan::{AstNode, AstSeparatedList, TextRange};
+use biome_rule_options::use_error_cause::UseErrorCauseOptions;
+
+declare_lint_rule! {
+    /// Disallow rethrowing caught errors without wrapping them.
+    ///
+    /// When rethrowing a caught error, it's recommended to wrap it in a new `Error` object to preserve the original error's stack trace and context.
+    /// The original error should be passed as the `cause` property of the new `Error` object.
+    ///
+    /// This rule enforces that practice, helping to maintain a clear and traceable error propagation chain, which is crucial for effective debugging.
+    ///
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// try {
+    ///   // ...
+    /// } catch (err) {
+    ///   throw new Error(err.message);
+    /// }
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// try {
+    /// 	doSomething();
+    /// } catch {
+    /// 	throw new Error("Something went wrong");
+    /// }
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// try {
+    ///   // ...
+    /// } catch ({ message }) {
+    ///   throw new Error(message);
+    /// }
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```js
+    /// try {
+    ///   // ...
+    /// } catch (err) {
+    ///   throw new Error("Something went wrong", { cause: err });
+    /// }
+    /// ```
+    ///
```

Comment:
The documentation doesn't explain and test the option `requireCatchParameter`. See the format for options: https://github.com/biomejs/biome/blob/main/crates/biome_analyze/CONTRIBUTING.md#general-structure